### PR TITLE
Add rPDU2BankStatusTable to generator example.

### PR DESCRIPTION
**Pull request by SuperQ**
Also updates with latest keepalived MIB.

Closes: https://github.com/prometheus/snmp_exporter/issues/375
Closes: https://github.com/prometheus/snmp_exporter/issues/383

Signed-off-by: Ben Kochie <superq@gmail.com>

### CODE REVIEW

**Comment by brian-brazil** [comment]
The generator test is failing.

**Comment by SuperQ** [comment]
I get different number of parse errors on my machine. Very strange.

**Comment by brian-brazil** [comment]
Did you do MIBDIRS=mibs?

**Comment by SuperQ** [comment]
Yes, I did, but the whole mibs dir was a mess. I've re-done the way we fetch Cisco's broken set of MIBs by only copying what we need out of it.

**Comment by SuperQ** [comment]
Ok, now the generator parse_errors runs without any errors.

File: generator/Makefile [hunk 1]
```diff
@@ -73,19 +89,45 @@ $(MIBDIR)/ARISTA-SW-IP-FORWARDING-MIB:
...
-
-$(MIBDIR)/KEEPALIVED-MIB.txt:
-	@echo ">> Downloading keepalived"
-	@curl -s -o $(MIBDIR)/KEEPALIVED-MIB.txt -L $(KEEPALIVED_URL)
-
-$(MIBDIR)/mikrotik.mib:
-	@echo ">> Downloading mikrotik"
-	@curl -s -o $(MIBDIR)/mikrotik.mib -L $(MIKROTIK_URL)
+		| tar -C $(MIBDIR)/cisco_v2 --strip-components=3 -zxv
+	cp mibs/cisco_v2/AIRESPACE-REF-MIB.my mibs/AIRESPACE-REF-MIB
```
Comment (brian-brazil) [inline generator/Makefile]: A for loop would be a bit clearer here I think

File: generator/Makefile [hunk 2]
```diff
@@ -34,18 +37,31 @@ all: mibs
...
 		$(MIBDIR)/.synology
 
+generator:
+	go build
+
+generate: generator mibs
+	MIBDIRS=$(MIB_PATH) ./generator generate
+
+parse_errors: generator mibs
+	MIBDIRS=$(MIB_PATH) ./generator parse_errors
```
Comment (brian-brazil) [inline generator/Makefile]: You know this will exit 0 even if there are parse errors?

File: generator/Makefile (hunk 1 above)
Comment (SuperQ) [reply generator/Makefile]: Maybe, I wanted to just keep it simple.

File: generator/Makefile (hunk 2 above)
Comment (SuperQ) [reply generator/Makefile]: Yup, it's just a shortcut for manual testing and visibility at this point. I was going to open an issue to add a "fail on errors" flag to the generator so we can include it as an actual part of the tests.

Comment (SuperQ) [reply generator/Makefile]: I added the flag to allow failure code on parse errors.

File: generator/main.go [hunk 3]
```diff
@@ -98,10 +99,15 @@ func main() {
 	command := kingpin.Parse()
 
 	parseErrors := initSNMP()
+	parseErrorCount := len(parseErrors)
 
-	if len(parseErrors) != 0 {
+	if parseErrorCount != 0 {
 		log.Warnf("NetSNMP reported %d parse errors", len(strings.Split(parseErrors, "\n")))
 	}
+	if !*failOnParseErrors {
```
Comment (brian-brazil) [inline generator/main.go]: I'd expect the process to exit here

File: generator/main.go [hunk 4]
```diff
@@ -124,4 +130,5 @@ func main() {
 				n.Oid, n.Label, t, n.TextualConvention, n.Hint, n.Indexes, implied, n.EnumValues, n.Description)
 		})
 	}
+	os.Exit(parseErrorCount)
```
Comment (brian-brazil) [inline generator/main.go]: This is unsafe, what if there's 256 parse errors?

File: generator/main.go [hunk 5]
```diff
@@ -86,6 +86,7 @@ func generateConfig(nodes *Node, nameToNode map[string]*Node) {
 }
 
 var (
+	failOnParseErrors  = kingpin.Flag("fail-on-parse-errors", "Exit with a non-zero status if there are parse errors").Default("false").Bool()
```
Comment (brian-brazil) [inline generator/main.go]: I'd put MIB in here somewhere, as it could be generator.yml you're talking about.

File: generator/main.go (hunk 3 above)
Comment (SuperQ) [reply generator/main.go]: I thought about that, but then it exits without printing the actual parse errors.

File: generator/main.go (hunk 4 above)
Comment (SuperQ) [reply generator/main.go]: Ahh, good point. I'll fix.

File: generator/main.go (hunk 5 above)
Comment (SuperQ) [reply generator/main.go]: Ok.

**Pull request merged**